### PR TITLE
RESINDEX-1323 install lib and bin to standard unix location

### PR DESCRIPTION
**Pull request by Liek0s**
Update CMakeLists to install to urlfactory.a to /usr/local/lib/liburlfactory.a and the tests to /usr/local/bin/

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,62 +1,82 @@
 cmake_minimum_required(VERSION 3.10)
+project(urlfactory)
 
-set(CMAKE_CXX_COMPILER /usr/bin/g++)
-set(CMAKE_C_COMPILER /usr/bin/gcc)
+set(CMAKE_CXX_STANDARD  11)
+set(CMAKE_C_COMPILER    /usr/bin/gcc)
+set(CMAKE_CXX_COMPILER  /usr/bin/g++)
```
Comment (dkurzaj) [inline CMakeLists.txt]: Use clang++ instead: `/usr/bin/clang++`

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,62 +1,82 @@
 cmake_minimum_required(VERSION 3.10)
+project(urlfactory)
 
-set(CMAKE_CXX_COMPILER /usr/bin/g++)
-set(CMAKE_C_COMPILER /usr/bin/gcc)
+set(CMAKE_CXX_STANDARD  11)
+set(CMAKE_C_COMPILER    /usr/bin/gcc)
```
Comment (dkurzaj) [inline CMakeLists.txt]: Use clang instead: `/usr/bin/clang`

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,62 +1,82 @@
 cmake_minimum_required(VERSION 3.10)
+project(urlfactory)
 
-set(CMAKE_CXX_COMPILER /usr/bin/g++)
-set(CMAKE_C_COMPILER /usr/bin/gcc)
+set(CMAKE_CXX_STANDARD  11)
```
Comment (dkurzaj) [inline CMakeLists.txt]: Use `17` instead

File: CMakeLists.txt [hunk 4]
```diff
@@ -1,62 +1,82 @@
...
+    DESTINATION lib/)
 
-install(TARGETS 
-tests
-urlfactory
-hexencoder
-robots
-urlparser
-DESTINATION /build/)
+install(TARGETS
```
Comment (jprenaudet) [inline CMakeLists.txt]: We do not need to install these test binaries

**Pull request merged**